src/server/scripts/Kalimdor/zone_orgrimmar.cpp

```diff
@@ -263,8 +264,579 @@ class npc_thrall_warchief : public CreatureScript
+                            felika->AI()->Talk(SAY_FELIKA_1);
+                        events.ScheduleEvent(EVENT_SCENE_4, 4500);
+                        break;
+                    case EVENT_SCENE_4:
+                        if (Creature* thathung = ObjectAccessor::GetCreature(*me, thungGUID))
+                            thathung->AI()->Talk(SAY_THATHUNG);
+                        events.ScheduleEvent(EVENT_SCENE_5, 4500);
+                        break;
+                    case EVENT_SCENE_5:
+                        if (Creature* sana = ObjectAccessor::GetCreature(*me, sanaGUID))
+                            sana->AI()->Talk(SAY_SANA);
+                        events.ScheduleEvent(EVENT_SCENE_6, 4500);
+                        break;
+                    case EVENT_SCENE_6:
+                        if (Creature* gryshka = ObjectAccessor::GetCreature(*me, gryshkaGUID))
+                            gryshka->AI()->Talk(SAY_GRYSHKA_2);
+                        events.ScheduleEvent(EVENT_SCENE_7, 4500);
+                        break;
+                    case EVENT_SCENE_7:
+                        if (Creature* kaja = ObjectAccessor::GetCreature(*me, kajaGUID))
+                            kaja->AI()->Talk(SAY_KAJA);
+                        events.ScheduleEvent(EVENT_SCENE_8, 4500);
+                        break;
+                    case EVENT_SCENE_8:
+                        if (Creature* felika = ObjectAccessor::GetCreature(*me, felikaGUID))
+                            felika->AI()->Talk(SAY_FELIKA_2);
+                        events.ScheduleEvent(EVENT_SCENE_9, 4500);
+                        break;
+                    case EVENT_SCENE_9:
+                        if (Creature* olvia = ObjectAccessor::GetCreature(*me, olviaGUID))
+                            olvia->AI()->Talk(SAY_OLVIA_2);
+                        events.ScheduleEvent(EVENT_SCENE_10, 4500);
+                        break;
+                    case EVENT_SCENE_10:
+                        Talk(SAY_RUNTHAK_1);
+                        events.ScheduleEvent(EVENT_SCENE_11, 1500);
+                        break;
+                    case EVENT_SCENE_11:
+                        Talk(SAY_RUNTHAK_2);
+                        events.ScheduleEvent(EVENT_SCENE_12, 4500);
+                        break;
+                    case EVENT_SCENE_12:
+                        Talk(SAY_RUNTHAK_3);
+                        events.ScheduleEvent(EVENT_SCENE_13, 4500);
+                        break;
+                    case EVENT_SCENE_13:
+                        Talk(SAY_RUNTHAK_4);
+                        events.ScheduleEvent(EVENT_RESET, 25000);
+                        break;
+                    case EVENT_RESET:
+                        Initialize();
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+    private:
+        EventMap events;
+        std::list<Unit*> citizenList;
```

Comment:
oh come on, do not store pointers, use GUIDs or in this case it looks like local variable should be enough